forum/profiles/permissions.py

```diff
@@ -22,3 +25,24 @@ def has_object_permission(self, request, view, obj):
         # Write permissions are only allowed to the owner of the object
         return obj.user == request.user
 
+
+class AbstractRolePermission(BasePermission):
+    role = None
+
+    def has_permission(self, request, view):
+        if request.method in SAFE_METHODS:
+            return True
+        try:
+            token_role_value = request.auth.get('role')
+        except AttributeError:
+            raise AuthenticationFailed("Invalid token")
+
+        return int(token_role_value) == self.role
```

Comment:
If `request.auth` is `None` or malformed, accessing `.get('role')` will raise an `AttributeError`. While this is partially handled, the code could include a more robust check. Consider adding an explicit check for the existence of `request.auth`.
```
def has_permission(self, request, view):
    if request.method in SAFE_METHODS:
        return True
    if not request.auth or not hasattr(request.auth, "get"):
        raise AuthenticationFailed("Invalid or missing token")
    token_role_value = request.auth.get('role')
    return int(token_role_value) == self.role
```